index.bs

```diff
@@ -349,6 +349,218 @@ One example of a feature that should be limited to secure contexts
+
+As an inherited property,
+'text-decoration' would draw underlines (or other decorations)
+in the style and position of each piece of text.
+As a non-inherited property, it would draw underlines
+in the style and position of the element that specifies 'text-decoration'.
+These behaviors lead to different results
+when the element that has 'text-decoration' declared
+has descendants that are at a different vertical-alignment (e.g., superscripts),
+in a different color,
+or in a different font size.
+
+The property was designed as not inherited
+because the behavior of having underlines
+have a consistent color, thickness, and position
+was considered the better behavior.
+
+However, in the end, some of the characteristics of the other behavior
+(such as its interaction with 'visibility' or text selection)
+were also seen as desirable, and it ended up being specified in a somewhat hybrid way
+that does involve the "code smell"
+of walking up the ancestor chain looking for text decorations on ancestors.
+</div>
+
+<h3 id="css-computed-value">Choose the computed value type based on how the property should inherit</h3>
+
+Another important choice in the design of every CSS property
+is what its <a>computed value</a> is.
+This is an abstract description of the value that is inherited when the property inherits.
+This is most important for inherited properties, but it does matter for all properties.
+
+Due to a somewhat unfortunate history,
+this is *not* the value that is returned by {{getComputedStyle()}},
+which is actually the <a>resolved value</a>,
+which can be the result of further computation starting from the computed value.
+
+The definition of a CSS property often has interactions with other CSS properties.
+Whether those interactions affect the property's computed value is a design choice,
+and it affects the behavior of inheritance.
+
+<a href="https://wiki.csswg.org/spec/property-dependencies">The dependencies
+that do affect computed values</a>
+are required not be circular.
+
+Making decisions about whether dependencies should affect computed values
+often involves tradeoffs of complexity.
+Sometimes it is less complex
+for the behavior to happen as part of the computation of the computed value;
+sometimes it is less complex 
+for the behavior to happen in whatever handles the computed value.
+(This complexity might be in the specification, in implementations, or both.
+Complexity in implementations might affect how fast things run for end users.)
+
+<div class="example">
+For example, the 'line-height' property accepts values that are integers,
+such as ''line-height: 1.4''.
+This value is a multiple of the 'font-size',
+so if the 'font-size' is ''20px'' then this would mean the line height is ''28px''.
+
+However, the computed value in this case is the number ''1.4'', not the length ''28px''.
+This is important because the value can inherit to elements that have a different font size:
```

Comment:
```suggestion
This is important because the value can inherit to elements that have a different font size,
and preserving the "multiple of the current element's font" behavior would be broken
if it turned into a length at computed-value time:
```